app/server/appsmith-server/src/test/java/com/appsmith/server/services/PageServiceTest.java

```diff
@@ -385,6 +385,103 @@ public void reuseDeletedPageName() {
                 .verifyComplete();
     }
 
+    @Test
+    @WithUserDetails(value = "api_user")
+    public void reOrderPageMovingUp() {
+
+        PageDTO testPage = new PageDTO();
+        testPage.setName("PageReorder TestApp1");
+        setupTestApplication();
+        testPage.setApplicationId(application.getId());
+        testPage.setOrder(0);
+        String pageId1 = testPage.getId();
+
+        applicationPageService.createPage(testPage);
+
+        testPage = new PageDTO();
+        testPage.setName("PageReorder TestApp2");
+        testPage.setApplicationId(application.getId());
+        testPage.setOrder(1);
+        String pageId2 = testPage.getId();
+
+        applicationPageService.createPage(testPage);
+
+        testPage = new PageDTO();
+        testPage.setName("PageReorder TestApp3");
+        testPage.setApplicationId(application.getId());
+        testPage.setOrder(2);
+        String pageId3 = testPage.getId();
+
+        applicationPageService.createPage(testPage);
+
+        Mono<Application> applicationMono = applicationPageService.reorderPage(application.getId(), pageId3, 0);
+        StepVerifier
+                .create(applicationMono)
+                .assertNext(application ->{
+                    final List<ApplicationPage> pages = application.getPages();
+                    for(ApplicationPage page : pages){
+                        if(pageId3.equals(page.getId())){
+                            assertTrue(page.getOrder() == 0);
```

Comment:
Let's use `assertThat` type of assertions since they report failing assertions in a much better way. The `assertTrue` will only tell that the condition is not true. But this will tell us that the expected value is `0`, but the actual value is `<something else>`. Makes it easier to troubleshoot failing tests.

```suggestion
                            assertThat(page.getOrder()).isEqualTo(0);
```